contracts/axelar-gateway/src/test_auth.rs

```diff
@@ -0,0 +1,545 @@
+
+    proof.signers = new_proof.signers;
+
+    // validate_proof should panic, signatures do not match signers
+    env.as_contract(&contract_id, || {
+        auth::validate_proof(&env, msg_hash, proof);
+    })
+}
+
+#[test]
+#[should_panic(expected = "Error(Contract, #8)")]
+fn fail_validate_proof_threshold_not_met() {
+    let (env, contract_id, client) = setup_env();
+    let user = Address::generate(&env);
+
+    let signers = initialize(&env, &client, user, randint(0, 10), randint(1, 10));
+
+    let env = &env;
+    let mut total_weight = 0u128;
+
+    let msg_hash = generate_random_payload_and_hash(env);
+    let mut proof = generate_proof(env, msg_hash.clone(), signers);
+
+    // Modify signatures to make them invalid
+    let mut new_signers = Vec::new(env);
+    for ProofSigner { signer, signature } in proof.signers.iter() {
+        total_weight += signer.weight;
+
+        if total_weight < proof.threshold {
+            new_signers.push_back(ProofSigner { signer, signature });
+        } else {
+            new_signers.push_back(ProofSigner {
+                signer,
+                signature: ProofSignature::Unsigned,
+            });
+        }
+    }
+    proof.signers = new_signers;
+
+    // should panic, all signatures are valid but total weight is below threshold
+    env.as_contract(&contract_id, || {
+        auth::validate_proof(&env, msg_hash, proof);
+    })
+}
+#[test]
+#[should_panic(expected = "Error(Contract, #6)")]
+fn fail_validate_proof_threshold_overflow() {
+    let (env, contract_id, client) = setup_env();
+    let user = Address::generate(&env);
+
+    let mut signers = initialize(&env, &client, user, randint(0, 10), randint(1, 10));
+
+    let env = &env;
+
+    let last_index = signers.signers.signers.len() - 1;
+
+    // get last signer and modify its weight to max u128 - 1
+    if let Some(mut last_signer) = signers.signers.signers.get(last_index) {
+        last_signer.weight = u128::MAX - 1;
+        signers.signers.signers.set(last_index, last_signer);
+    }
+
+    let msg_hash = generate_random_payload_and_hash(env);
+    let proof = generate_proof(env, msg_hash.clone(), signers);
+
+    // should panic, as modified signer wouldn't match the epoch
+    env.as_contract(&contract_id, || {
+        auth::validate_proof(&env, msg_hash, proof);
+    });
+}
+
+#[test]
+fn test_rotate_signers() {
+    let (env, contract_id, client) = setup_env();
+
+    let user = Address::generate(&env);
+    let previous_signer_retention = 1;
+
+    let signers = initialize(
+        &env,
+        &client,
+        user.clone(),
+        previous_signer_retention,
+        randint(1, 10),
+    );
+
+    let msg_hash = generate_random_payload_and_hash(&env);
+
+    let new_signers = generate_signers_set(&env, randint(1, 10), signers.domain_separator);
+
+    testutils::rotate_signers(&env, &contract_id, new_signers.clone());
+
+    assert_invocation(
```

Comment:
remove `assert_invocation` here since there's no `require_auth()` call in `auth:rotate_signers`

Reply:
this solves the issue of the test not passing, nice catch